druid/src/widget/flex.rs

```diff
@@ -416,7 +429,7 @@ impl<T: Data> Flex<T> {
     ///
     /// Convenient for assembling a group of widgets in a single expression.
     pub fn with_child(mut self, child: impl Widget<T> + 'static) -> Self {
-        self.add_flex_child(child, 0.0);
+        self.add_child(child);
```

Comment:
Ah so, if it's negative or 0.0 I should make it a non flex child? Would there still be a warning about invalid flex value?

Reply:
I'm just trying to think this through.

My basic thinking is that flex values are relative, and we have no real way of knowing that `1.0` is a useful or desirable value, and so we could go non-flex instead. I *think* that most people who would pass `0.0` *probably* want the non-flex behaviour? 

I could really go either way. One thing I would probably do is note in the log message that maybe they want `add_child` instead of `add_flex_child`, or whatever, and also a pointer to the Flex docs.

okay, I say: if flex is `0.0` add a non-flex child, and print  the following:

"Flex value should be > 0.0. To add a non-flex child use the add_child or with_child methods. See the Flex docs for more information."

Reply:
Ok yeah, I see your point. It's hard to know what the appropriate default flex value would be for that Flex widget. I'll do that. You said to point to the flex docs for info, but I feel like it's kind of barely mentioned in the reading. 
>Unlike a non-flex child, a child with a non-zero flex factor has a maximum allowed size on the main axis;

Wish there was a way to directly link that part. If this is somehow in the future being ignored by people after the warning then it's only a doc comment fix away to make it more transparent.

Anyways, I'll add these changes.